components/Input/Address/AddressPicker/index.tsx

```diff
@@ -0,0 +1,229 @@
+import { Exchange } from "../../../../Models/Exchange";
+import AddressBook from "./AddressBook";
+import AddressButton from "./AddressButton";
+import { useQueryState } from "../../../../context/query";
+
+export enum AddressGroup {
+    ConnectedWallet = "Connected wallet",
+    ManualAdded = "Added Manually",
+    RecentlyUsed = "Recently used",
+    FromQuery = "Partner",
+}
+
+export type AddressItem = {
+    address: string,
+    group: AddressGroup,
+    networkType?: NetworkType | string
+    date?: string
+}
+
+interface Input extends Omit<React.HTMLProps<HTMLInputElement>, 'ref' | 'as' | 'onChange'> {
+    showAddressModal: boolean;
+    setShowAddressModal: (show: boolean) => void;
+    hideLabel?: boolean;
+    disabled: boolean;
+    name: string;
+    children?: JSX.Element | JSX.Element[];
+    ref?: any;
+    close: () => void,
+    partner?: Partner,
+    canFocus?: boolean,
+    address_book?: AddressBookItem[],
+}
+
+const AddressPicker: FC<Input> = forwardRef<HTMLInputElement, Input>(function Address
+    ({ showAddressModal, setShowAddressModal, name, canFocus, close, address_book, disabled, partner }, ref) {
+
+    const {
+        values,
+        setFieldValue
+    } = useFormikContext<SwapFormValues>();
+    const query = useQueryState()
+    const { destination_address, to: destination, toExchange: destinationExchange, toCurrency: destinationAsset } = values
+
+    const { disconnectWallet, getAutofillProvider: getProvider } = useWallet()
+    const provider = useMemo(() => {
+        return values?.to && getProvider(values?.to)
+    }, [values?.to, getProvider])
+    const connectedWallet = provider?.getConnectedWallet()
+    const connectedWalletAddress = connectedWallet?.address
+
+    const [isConnecting, setIsConnecting] = useState(false)
+    const [manualAddress, setManualAddress] = useState<string>('')
+    const [newAddress, setNewAddress] = useState<{ address: string, networkType: NetworkType | string } | undefined>()
+    const [wrongNetwork, setWrongNetwork] = useState(false)
+
+    const inputReference = useRef<HTMLInputElement>(null);
+
+    const menuItems = destination && generateMenuItems({ address_book, destination, destinationExchange, connectedWalletAddress, newAddress, addressFromQuery: query.destAddress, partner })
+    const filteredAddresses = menuItems?.filter(a => a.group !== AddressGroup.ConnectedWallet)
+    const destinationAddressItem = destination && menuItems?.find(a => addressFormat(a.address, destination) === addressFormat(destination_address || '', destination))
+
+    const handleSelectAddress = useCallback((value: string) => {
+        const address = destination && menuItems?.find(a => addressFormat(a.address, destination) === addressFormat(value, destination))?.address
+        setFieldValue("destination_address", address)
+        close()
+    }, [close, setFieldValue])
+
+    const autofillConnectedWallet = useCallback(() => {
+        //TODO move to wallet implementation
```

Comment:
Isn't this done somewhere? sort out

Reply:
https://github.com/layerswap/layerswapapp/pull/924/files